muspinsim_config/config_macros.xml

```diff
@@ -142,4 +142,95 @@
             </when>
         </conditional>
     </xml>
+    <xml name="spin_options">
+        <repeat name="spins" title="Spins to simulate" min="1" help="Specify the spins to be used in the system. This should include a muon (mu) and one or more electrons (e)">
+            <conditional name="spin_options">
+                <param name="spin_preset" type="select" value="mu" label="Species" help="Select 'custom' to define own">
+                    <option selected="true" value="mu">mu</option>
+                    <option value="e">e</option>
+                    <option value="custom">custom</option>
+                </param>
+                <when value="custom">
+                    <param name="spin" optional="false" type="text" label="Species name"/>
+                    <param name="atomic_mass" optional="true" type="integer" min="0" value="" label="Atomic mass" help="Leave blank to use default mass - whole numbers only"/>
+                </when>
+                <when value="mu"/>
+                <when value="e"/>
+            </conditional>
+        </repeat>
+    </xml>
+    <xml name="additional_spin_options">
+        <repeat name="spins" title="Additional spins to simulate" min="0" help="Specify any additional spins to be used in the system other than those generated from the structure file.">
```

Comment:
Having played with this a bit in the UI, I think we should expand the help text (either here, or maybe in `generator_options`) to be more explicit about the how the spins are written to file by the generator. The points I think we should cover:

- The muon will always have index `1`
- Following this, have the `N` closest spins (ordered by distance?)
- The first additional spin will therefore have index `N+2`, e.g. `6` for the default case

Obviously it's not going to be easy to say add an interaction with the closest neighbour (`2`) and an electron you added (since it's not clear which atom is going to be closest). But hopefully more explanation would make it easier to say add an interaction with an additional spin and the muon (by making the index of each clear). Like I said I'm not sure what the best place/wording for this is - what do you think?

Reply:
I agree although I think its best under the interactions section heading, after adding any extra spins. Will update and see what it looks like.